rs/validator/ingress_message/tests/validate_request.rs

```diff
@@ -273,6 +277,32 @@ mod ingress_expiry {
             assert_matches!(result, Ok(()), "Test with {builder_info} failed");
         }
     }
+
+    #[test]
+    fn should_not_error_when_anonymous_read_state_request_expired() {
+        let verifier = verifier_at_time(CURRENT_TIME).build();
+        let request = HttpRequestBuilder::new_read_state()
+            .with_authentication(AuthenticationScheme::Anonymous)
+            .with_ingress_expiry_at(CURRENT_TIME.saturating_sub(Duration::from_nanos(1)))
```

Comment:
nit: Would it be meaningful in this test and the one below to test that the other authentication schemes on that value for `ingress_expiry_at` do fail verification due to being expired?

Reply:
I didn't check the whole code base but I'd assume that the verification for other authentication schemes is checked in some other tests. If it is missing, then I agree that it should be added (but then the test function name must be updated or the checks are performed in additional tests).

Reply:
@gregorydemay, a few lines above (in the `ingress_expiry` module in this file here), there is (as @THLO assumed) `should_error_when_request_expired`, which tests that verification fails for expired updated update calls, for expired non-anonymous queries, and for expired non-anonymous read-state requests. (That test uses the same `ingress_expiry_at` value (CURRENT_TIME-1) but that shouldn't really matter.)

Do you think that is sufficient?

There are also the related `should_error_when_request_expiry_too_far_in_future` and `should_accept_request_when_expiry_within_acceptable_bounds`.

Reply:
Thanks @fspreiss for looking into it! I suggested this to only make it clear that in your current test `should_not_error_when_anonymous_read_state_request_expired` the requests are indeed expired for the other authentication schemes. It's also ok to leave as is if you think it's overkill.

Reply:
Given that `should_error_when_request_expired` tests exactly that (even with exactly the same `ingress_expiry_at` value CURRENT_TIME-1), I'd say the current tests are sufficient.